Chessington.GameEngine/Pieces/Piece.cs

```diff
@@ -34,27 +34,49 @@ public IEnumerable<Square> GetDiagonalMoves(Board board)
                 {
                     availableMoves.Add(Square.At(currentRow + columnsAway, col));
                 }
+
                 if (Enumerable.Range(0, 8).Contains(currentRow - columnsAway))
                 {
                     availableMoves.Add(Square.At(currentRow - columnsAway, col));
                 }
             }
+
             availableMoves.RemoveAll(x => x == Square.At(currentRow, currentCol));
-            return availableMoves; 
+            return availableMoves;
         }
 
         public IEnumerable<Square> GetLateralMoves(Board board)
+        {
+            List<Square> availableMoves = new List<Square>();
+
+            availableMoves.AddRange(ExploreInOneDirection(1, 0, board));
+            availableMoves.AddRange(ExploreInOneDirection(-1, 0, board));
+            availableMoves.AddRange(ExploreInOneDirection(0, 1, board));
+            availableMoves.AddRange(ExploreInOneDirection(0, -1, board));
+
+            availableMoves.OrderBy(x => x.Row).OrderByDescending(x => x.Col);
+
+            return availableMoves;
+        }
+
+        public IEnumerable<Square> ExploreInOneDirection(int rowDirection, int colDirection, Board board)
         {
             var currentPos = board.FindPiece(this);
-            var row = currentPos.Row;
-            var col = currentPos.Col;
+            var currentRow = currentPos.Row;
+            var currentCol = currentPos.Col;
             List<Square> availableMoves = new List<Square>();
-            foreach (var i in Enumerable.Range(0, 8))
+
+            int tileDistance = 1;
+            while (Square.At(currentRow + tileDistance * rowDirection, currentCol + tileDistance * colDirection)
```

Comment:
maybe you can combine Inbound and IsOccupied into one method IsValid?

Reply:
This has been changed in the next pull request (10-12)